sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -310,6 +310,10 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
       case logical.MapPartitionsInR(f, p, b, is, os, objAttr, child) =>
         execution.MapPartitionsExec(
           execution.r.MapPartitionsRWrapper(f, p, b, is, os), objAttr, planLater(child)) :: Nil
+      case logical.MapGroupsInR(f, p, b, is, os, key, value, grouping, data, objAttr,
+        child) => execution.MapGroupsExec(
+          execution.r.MapGroupsRWrapper(f, p, b, is, os), key, value, grouping,
```

Comment:
don't use MapGroupsRWrapper, as dapply() does. Because this way, the Wrapper function will be called multiple times for a partition, which means R worker will be launched multiple times, which hurts performance.